internal/websocket/ws.go

```diff
@@ -330,14 +321,11 @@ func (ws *Websocket) dial(ctx context.Context, mode ipMode) (*nhws.Conn, *http.R
 	if err != nil {
 		return nil, nil, err
 	}
-
+	ws.updateTransportDialContext(mode)
 	conn, resp, err := nhws.Dial(ctx, url,
```

Comment:
This is the place the the final `*http.Client` should be created.  It should not be stored, just passed to the `Dial()` function.